src/message.rs

```diff
@@ -62,3 +64,57 @@ impl fmt::Display for Message {
         )
     }
 }
+
+impl Message {
+    pub fn to_annotated_source(&self) -> String {
+        let source = fs::read_to_string(&self.filename).unwrap();
+        let error_lines = source
+            .lines()
+            .skip(self.location.row() - 1)
+            .take(self.end_location.row() - self.location.row() + 1)
+            .collect::<Vec<_>>();
+        let range_end = if self.location.row() == self.end_location.row() {
+            self.end_location.column()
+        } else {
+            error_lines
+                .iter()
+                .enumerate()
+                .map(|(i, line)| {
+                    if i == error_lines.len() - 1 {
+                        self.end_location.column()
+                    } else {
+                        line.len() + 1
+                    }
+                })
+                .sum()
+        };
+        let body = self.kind.body();
+        let code = self.kind.code().as_ref();
+        let rel_path = relativize_path(Path::new(&self.filename));
+        let source = error_lines.join("\n");
+        let snippet = Snippet {
+            title: Some(Annotation {
+                label: Some(&body),
+                id: Some(code),
+                annotation_type: AnnotationType::Error,
+            }),
+            footer: vec![],
```

Comment:
This is where we can show notes or help messages:

```diff
-            footer: vec![],
+            footer: vec![Annotation {
+                label: Some("Add --fix to automatically this error"),
+                id: None,
+                annotation_type: AnnotationType::Help,
+            }],
```

<img width="520" alt="image" src="https://user-images.githubusercontent.com/17039389/201500272-700df862-9541-4962-ad80-6ac4c09328bd.png">